x-pack/plugins/cloud_security_posture/public/components/csp_page_template.tsx

```diff
@@ -0,0 +1,175 @@
+      dashboard: allNavigationItems.dashboard,
+      findings: allNavigationItems.findings,
+      benchmark: allNavigationItems.benchmarks,
+    }),
+  },
+  restrictWidth: false,
+};
+
+export const DEFAULT_NO_DATA_CONFIG: KibanaPageTemplateProps['noDataConfig'] = {
+  pageTitle: DEFAULT_NO_DATA_TEXT.PAGE_TITLE,
+  solution: DEFAULT_NO_DATA_TEXT.SOLUTION,
+  // TODO: Add real docs link once we have it
+  docsLink: 'https://www.elastic.co/guide/index.html',
+  logo: 'logoSecurity',
+  actions: {},
+};
+
+const getPackageNotInstalledConfig = (
+  cisIntegrationLink: string
+): KibanaPageTemplateProps['noDataConfig'] => ({
+  pageTitle: PACKAGE_NOT_INSTALLED_TEXT.PAGE_TITLE,
+  solution: PACKAGE_NOT_INSTALLED_TEXT.SOLUTION,
+  // TODO: Add real docs link once we have it
+  docsLink: 'https://www.elastic.co/guide/index.html',
+  logo: 'logoSecurity',
+  actions: {
+    elasticAgent: {
+      href: cisIntegrationLink,
+      title: PACKAGE_NOT_INSTALLED_TEXT.BUTTON_TITLE,
+      description: PACKAGE_NOT_INSTALLED_TEXT.DESCRIPTION,
+    },
+  },
+});
+
+const DefaultLoading = () => <CspLoadingState>{LOADING}</CspLoadingState>;
+
+const DefaultError = (error: QueryError) => (
+  <EuiEmptyPrompt
+    color="danger"
+    iconType="alert"
+    title={
+      <>
+        <EuiTitle>
+          <h2>
+            <FormattedMessage
+              id="xpack.csp.pageTemplate.loadErrorMessage"
+              defaultMessage="We couldn't fetch your cloud security posture data"
+            />
+          </h2>
+        </EuiTitle>
+        {error.body?.error && error.body?.statusCode && (
+          <EuiTitle size="xs">
+            <h5>{`${error.body.error} (${error.body.statusCode})`}</h5>
+          </EuiTitle>
+        )}
+        {error.body?.message && (
+          <EuiTitle size="xs">
+            <h5>{error.body.message}</h5>
+          </EuiTitle>
+        )}
+      </>
+    }
+  />
+);
+
+export const CspPageTemplate: React.FC<
+  KibanaPageTemplateProps & {
+    loadingRender?: () => JSX.Element;
+    errorRender?: (error: QueryError) => JSX.Element;
+    query?: UseQueryResult;
+  }
+> = ({
+  query,
+  children,
+  loadingRender = DefaultLoading,
+  errorRender = DefaultError,
+  ...kibanaPageTemplateProps
+}) => {
+  const cisKubernetesPackageInfo = useCisKubernetesIntegration();
+  const cisIntegrationLink = useCISIntegrationLink();
+
+  const getNoDataConfig = (): KibanaPageTemplateProps['noDataConfig'] => {
+    // TODO: add 'installed_failed' configurations
+    if (cisKubernetesPackageInfo?.status === 'not_installed') {
+      return getPackageNotInstalledConfig(cisIntegrationLink);
+    }
+
+    // when query was successful, but data is undefined
+    if (query && query.isSuccess && !query.data) {
+      return kibanaPageTemplateProps.noDataConfig || DEFAULT_NO_DATA_CONFIG;
+    }
+
+    // when the consumer didn't pass a query, most likely to handle the render on his own
+    if (!query) return kibanaPageTemplateProps.noDataConfig;
```

Comment:
seems like both `if` blocks should return `kibanaPageTemplateProps.noDataConfig || DEFAULT_NO_DATA_CONFIG;` ?

Reply:
yep, if I won't have the second `if`, in case of no query passed we will get undefined, but what I want is to get the passed data from the property. defaulting is not an option since `undefined` is a legit value in this case which needs to be accounted for